ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Wallets.hs

```diff
@@ -175,3 +200,25 @@ updateHdWalletAssurance
     -> IO (Either HD.UnknownHdRoot HdRoot)
 updateHdWalletAssurance pw hdRootId assuranceLevel =
     update' (pw ^. wallets) (UpdateHdWalletAssurance hdRootId assuranceLevel)
+
+{-------------------------------------------------------------------------------
+  Automatic creation of Wallet Account and Address
+-------------------------------------------------------------------------------}
+
+addAddress 
+    :: PassiveWallet
+    -> WalletId
+    -> PassPhrase
+    -> HdRoot
+    -> IO (Either CreateWalletError HdRoot)
+addAddress pw walletId passphrase hdRoot = do 
+    accountRes <- createAccount (AccountName "un-named") walletId pw
+    case accountRes of 
+        Left err      -> return . Left $ CreateAccountFailed err
+        Right account -> do
+            let accountId = AccountIdHdSeq $ account ^. HD.hdAccountId
+            addressRes <- createAddress passphrase accountId HdChainInternal pw
```

Comment:
`External` chain is [more suitable](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#Change) here. Addresses on the internal chain should only ever be generated as change outputs.